sygnal/gcmpushkin.py

```diff
@@ -300,13 +304,22 @@ async def _request_dispatch(self, n, log, body, headers, pushkeys, span):
     async def _dispatch_notification_unlimited(self, n, device, context):
         log = NotificationLoggerAdapter(logger, {"request_id": context.request_id})
 
+        # `_dispatch_notification_unlimited` gets called once for each device in the
+        # `Notification` with a matching app ID. We do something a little dirty and
+        # perform all of our dispatches the first time we get called for a
+        # `Notification` and do nothing for the rest of the times we get called.
+        app_id_pattern = glob_to_regex(self.name, ignore_case=False)
```

Comment:
Please can you pre-compile this at startup? `re.compile` ~~is~~ was(?)* expensive.

*I believe it used to bypass the cache. Looking at the code in my Python 3.9 install, it seems like it uses the cache now.
However, pre-compiling is probably still best practice, especially since it's trivial to do in this case.

Reply:
I've changed it to be explicitly cached on startup.